libobs/util/platform-nix-portal.c

```diff
@@ -117,41 +138,58 @@ static void do_inhibit(struct portal_inhibit_info *info, const char *reason)
 
 	bfree(token);
 
-	reply = g_dbus_connection_call_sync(
-		info->c, PORTAL_NAME, "/org/freedesktop/portal/desktop",
-		"org.freedesktop.portal.Inhibit", "Inhibit",
-		g_variant_new("(sua{sv})", "", flags, &options), NULL,
-		G_DBUS_CALL_FLAGS_NONE, -1, NULL, &error);
-
-	if (error != NULL) {
-		blog(LOG_ERROR, "Failed to inhibit: %s", error->message);
-		unsubscribe_from_request(info);
-		remove_inhibit_data(info);
-		return;
-	}
+	info->cancellable = g_cancellable_new();
+	g_dbus_connection_call(info->c, PORTAL_NAME,
+			       "/org/freedesktop/portal/desktop",
+			       "org.freedesktop.portal.Inhibit", "Inhibit",
+			       g_variant_new("(sua{sv})", "", flags, &options),
+			       NULL, G_DBUS_CALL_FLAGS_NONE, -1,
+			       info->cancellable, inhibited_cb, info);
 }
 
-static void do_uninhibit(struct portal_inhibit_info *info)
+static void uninhibited_cb(GObject *source_object, GAsyncResult *result,
+			   gpointer user_data)
 {
-	g_autoptr(GError) error = NULL;
+	UNUSED_PARAMETER(source_object);
 
-	g_dbus_connection_call_sync(info->c, PORTAL_NAME, info->request_path,
-				    "org.freedesktop.portal.Request", "Close",
-				    g_variant_new("()"), G_VARIANT_TYPE_UNIT,
-				    G_DBUS_CALL_FLAGS_NONE, -1, NULL, &error);
+	struct portal_inhibit_info *info = user_data;
+	g_autoptr(GVariant) reply = NULL;
+	g_autoptr(GError) error = NULL;
 
-	remove_inhibit_data(info);
+	reply = g_dbus_connection_call_finish(info->c, result, &error);
 
 	if (error)
 		blog(LOG_WARNING, "Error uninhibiting: %s", error->message);
 }
 
+static void do_uninhibit(struct portal_inhibit_info *info)
+{
+	if (info->cancellable) {
+		/* If uninhibit is called before the inhibit call is finished,
+		 * cancel it instead.
+		 */
+		g_cancellable_cancel(info->cancellable);
+		g_clear_object(&info->cancellable);
```

Comment:
Is it okay to clear this both here and in portal_inhibit_info_destroy?

Reply:
It is. `g_clear_*()` macros are handy precisely because they perform NULL checks :)